backend/src/roles/roles.service.ts

```diff
@@ -1,53 +1,84 @@
 import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+
+export class RoleDetails {
+  id!: string;
+  name!: string;
+  description?: string;
+  permissions?: string[];
+}
 
-import { RoleDetails } from './role-details.model';
 @Injectable()
 export class RolesService {
-  private roles: RoleDetails[] = [
-    {
-      id: 'r1',
-      name: 'editor',
-      description: 'Can edit content',
-      permissions: ['read', 'write'],
-    },
-    {
-      id: 'r2',
-      name: 'admin',
-      description: 'Full access',
-      permissions: ['read', 'write', 'delete', 'admin'],
-    },
-  ];
+  constructor(private prisma: PrismaService) {}
 
-  getAll(): RoleDetails[] {
-    return this.roles;
+  async getAll(): Promise<RoleDetails[]> {
+    const roles = await this.prisma.role.findMany();
+    return roles.map((r) => ({
+      id: r.id,
+      name: r.name,
+      description: r.description ?? '',
+      permissions: r.permissions ?? [],
+    }));
   }
 
-  getById(id: string): RoleDetails | undefined {
-    return this.roles.find((r) => r.id === id);
+  async getById(id: string): Promise<RoleDetails | undefined> {
+    const r = await this.prisma.role.findUnique({ where: { id } });
+    if (!r) return undefined;
+    return {
+      id: r.id,
+      name: r.name,
+      description: r.description ?? '',
+      permissions: r.permissions ?? [],
+    };
   }
 
-  create(role: Partial<RoleDetails>): RoleDetails {
-    const newRole: RoleDetails = {
-      id: role.id || Date.now().toString(),
-      name: role.name ?? '',
-      description: role.description ?? '',
-      permissions: role.permissions ?? [],
+  async create(role: Partial<RoleDetails>): Promise<RoleDetails> {
+    const created = await this.prisma.role.create({
+      data: {
+        name: role.name ?? '',
+        description: role.description ?? undefined,
+        permissions: role.permissions ?? [],
+      },
+    });
+    return {
+      id: created.id,
+      name: created.name,
+      description: created.description ?? '',
+      permissions: created.permissions ?? [],
     };
```

Comment:
No need of two step mapping, use return directly with await.

like: 

`return await this.prisma.role.create()`